int32 index
---
string spec
StopCondition stop_condition